taipy/core/taipy.py

```diff
@@ -141,6 +169,8 @@ def is_readable(
         return _JobManagerFactory._build_manager()._is_readable(entity)  # type: ignore
     if isinstance(entity, DataNode) or (isinstance(entity, str) and entity.startswith(DataNode._ID_PREFIX)):
         return _DataManagerFactory._build_manager()._is_readable(entity)  # type: ignore
+    if isinstance(entity, Submission) or (isinstance(entity, str) and entity.startswith(Submission._ID_PREFIX)):
+        return _SubmissionManagerFactory._build_manager()._is_readable(entity)  # type: ignore
```

Comment:
This condition is duplicate (lines 130 and 401). Can you create a dedicated function that better explains what this condition does?

Reply:
done :D